vpp/clientconfigsrv.go

```diff
@@ -0,0 +1,117 @@
+package vpp
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+type ClientContext struct {
+	HostName string `json:"hostname"`
+	Guid     string `json:"guid"`
+}
+
+type Location struct {
+	LocationName string `json:"locationName"`
+	LocationId   int    `json:"locationId"`
+}
+
+type ClientConfigSrvResponse struct {
+	ClientContext      interface{} `json:"clientContext"`
+	AppleId            string      `json:"appleId"`
+	OrganizationIdHash string      `json:"organizationIdHash"`
+	Status             int         `json:"status"`
+	OrganizationId     int         `json:"organizationId"`
+	UId                string      `json:"uId"`
+	CountryCode        string      `json:"countryCode"`
+	Location           Location    `json:"location"`
+	ApnToken           string      `json:"apnToken"`
+	Email              string      `json:"email"`
+}
+
+type ClientConfigSrv struct {
+	ClientContext      ClientContext `json:"clientContext"`
+	AppleId            string        `json:"appleId"`
+	OrganizationIdHash string        `json:"organizationIdHash"`
+	Status             int           `json:"status"`
+	OrganizationId     int           `json:"organizationId"`
+	UId                string        `json:"uId"`
+	CountryCode        string        `json:"countryCode"`
+	Location           Location      `json:"location"`
+	ApnToken           string        `json:"apnToken"`
+	Email              string        `json:"email"`
+}
+
+func (c *Client) GetClientConfigSrv() (*ClientConfigSrv, error) {
+	request := map[string]interface{}{
+		"sToken": c.sToken,
+	}
+
+	clientConfigSrvUrl := c.VPPServiceConfigSrv.ClientConfigSrvUrl
+	var response ClientConfigSrvResponse
+	req, err := c.newRequest("POST", clientConfigSrvUrl, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "create ClientConfigSrv request")
+	}
+
+	err = c.do(req, &response)
+
+	var context = response.ClientContext
```

Comment:
can you explain clientContext to me? Its not quite clear why it's named this way and what it does.

I see below you happen to sprintf a context too. Not sure I follow.

Reply:
I removed the sprintf line.
ClientContext is a string returned in the json from the ClientConfigSrv.  It associates a particular VPP token with your particular MDM.
Practically, if you send an InstallApplication command from your MDM to a device for a VPP app, the app won't install unless the ClientContext set in the VPP token matches the URL of your MDM server.